packages/vitest/src/node/cli-api.ts

```diff
@@ -30,7 +30,7 @@ export async function startVitest(
   process.env.VITEST = 'true'
   process.env.NODE_ENV ??= options.mode || 'test'
 
-  if (options.run)
+  if (options.run && !options.browser)
```

Comment:
Browser should not affect this logic in this place. This line is only about inverting `watch === !run`